<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="plot_cost" default="false"/>
    <arg name="description_file" default="$(find mppi_pathfinder)/resources/pathfinder.urdf"/>
    <!-- TF prefix -->
    <arg name="tf_prefix" default=""/>
    <!-- Load robot description -->
    <param name="robot_description" command="$(find xacro)/xacro $(arg description_file)"/>

    <!--  Run the test node-->
    <node name="pathfinder_control" type="pathfinder_control" pkg="mppi_pathfinder" output="screen">
        <rosparam command="load" file="$(find mppi_pathfinder)/config/params.yaml"/>
        <param name="config_file" value="$(find mppi_pathfinder)/config/params.yaml"/>
    </node>

    <!-- Publish robot state (joints) as TF -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static" value="true"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states" to="/joint_states"/>
    </node>

    <!-- Open Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_pathfinder)/config/config.rviz" output="screen"/>

    <group if="$(arg plot_cost)">
        <arg name="mp_config" default="$(find mppi_pathfinder)/resources/rqt_multiplot.xml"/>
        <node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot" args="--multiplot-config $(arg mp_config)"
              output="screen"/>
    </group>
</launch>
